fix(docs): update TodoList tutorial

dhmlau opened the pull request:
Fixes https://github.com/strongloop/loopback-next/issues/3742

Now that we have `lb4 relation` command, we need to update the TodoList tutorial to remove the manual steps because those are taken care by the relation generator command.

- I've shuffled around the content a bit so a good portion of it are existing content. 
- The TodoList tutorial also has the `hasOne` portion but since we don't have the `hasOne` option available on `lb4 relation` (see #2980), I didn't update the tutorial to include that.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

agnes512 commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -0,0 +1,141 @@
+### Define the model relation
+
+We are going to add the model relation to indicate the relation that `TodoList`
+_hasMany_ `Todo` using the `lb4 relation` command.
```
We can link `lb4 relation` to https://loopback.io/doc/en/lb4/Relation-generator.html

agnes512 commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -0,0 +1,141 @@
+? Source property name for the relation getter todos
+? Allow TodoList queries to include data from related Todo instances? Yes
+   create src/controllers/todo-list-todo.controller.ts
+
```
I'd like to add some explanations about the default foreign key name/source property names.

```
Notice: we use **default** foreign key and source property names in this case. 
If you'd like to customize them, please check `Relation Metadata` https://loopback.io/doc/en/lb4/HasMany-relation.html#relation-metadata 
and other relations as well.
```

agnes512 commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -0,0 +1,141 @@
+### Behind the scene
+
+If you want to understand the code changes introduced from the relation
+generator command, read on the details in this section; otherwise, you are ready
```
Suggest:
read on the details in `Relation Generator` https://loopback.io/doc/en/lb4/Relation-generator.html

dhmlau replied:
actually what i want to say is if you want to understand more about the code changed introduced from running the command, read this `Behind the scene` section, otherwise, just move on to the next step.  The `Relation Generator` page won't have that details.

agnes512 replied:
I see, make sense to me then 👍

agnes512 commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -0,0 +1,141 @@
+  'todos',
+  todoRepositoryGetter,
+);
+this.registerInclusionResolver('todos', this.todos.inclusionResolver);
```
Suggest:
add a comment 
// this line enables inclusion for this relation

dhmlau replied:
added it for both hasMany and belongsTo code snippet

agnes512 commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -0,0 +1,141 @@
+  'todoList',
+  todoListRepositoryGetter,
+);
+this.registerInclusionResolver('todoList', this.todoList.inclusionResolver);
```
Suggest:
add a comment 
// this line enables inclusion for this relation

agnes512 commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -0,0 +1,141 @@
+
+### Navigation
+
+Previous step: [Add TodoList repository](todo-list-tutorial-repository.md)
```
`[Add TodoList Repository]`

agnes512 commented on docs/site/tutorials/todo-list/todo-list-tutorial-repository.md:
```diff
@@ -37,115 +37,8 @@ bridge)
 Previous step: [Add TodoList model](todo-list-tutorial-model.md)
 
-Last step: [Add TodoList controller](todo-list-tutorial-controller.md)
+Last step: [Add model relations](todo-list-tutorial-relations.md)
```
`[Add Model Relations]`

agnes512 commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -0,0 +1,141 @@
+#### Inclusion of Related Models
+
+When running the `lb4 relation` command, the prompt below registers the
+inclusionResolver for this belongsTo relation for you.
```
inclusionResolver -> `inclusionResolver`

emonddr replied:
Here it seems we are asking user to run it again.

> When running the `lb4 relation` command, the prompt below registers the
inclusionResolver for this belongsTo relation for you. 

We need to rewrite this paragraph in the proper tense.

Perhaps something like: 

When we ran the `lb4 relation` commands above,  we accepted the default of `Yes` to the prompt:

```sh
? Allow Order queries to include data from related Customer instances? (Y/n)
```

This registered the inclusionResolver for the relation(s) you were working with above.

Make sure to chose `yes` ...

emonddr commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -0,0 +1,141 @@
+
+#### Relation decorators
+
+When specifying the `hasMany` relation, it adds `@hasMany()` decorator defines
```
> When specifying the `hasMany` relation, it adds `@hasMany()` decorator defines this property.

What are we trying to say here?

dhmlau replied:
trying to say when running the `lb4 relation` command, the decorator was added to the property. Let me do some rewording here.

agnes512 commented on docs/site/sidebars/lb4_sidebar.yml:
```diff
@@ -318,6 +318,10 @@ children:
       url: todo-list-tutorial-model.html
       output: 'web, pdf'
 
+    - title: 'Add Model Relations'
```
yes

jannyHou replied:
+1 'Add Model Relations' should be after 'Add TodoList Repository'

jannyHou commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -0,0 +1,141 @@
+
+On the other end,
+[`BelongsToAccessor`](https://loopback.io/doc/en/lb4/apidocs.repository.belongstoaccessor.html)
+also comes with an inclusion resolver property that we can register on the
```
nitpick: I feel a more accurate word is `function` instead of `property`.

dhmlau replied:
changed it to `function`.

emonddr commented on docs/site/tutorials/todo-list/todo-list-tutorial-relations.md:
```diff
@@ -48,9 +56,9 @@ to move to the next step to create the controller.
-When specifying the `hasMany` relation, it adds `@hasMany()` decorator defines
-this property. As the decorator's name suggests, `@hasMany()` informs LoopBack 4
-that a todo list can have many todo items.
+When we added the `hasMany` relation using the `lb4 relation` command, it added
```
> When we added the `hasMany` relation using the `lb4 relation` command, it added
`@hasMany()` decorator defines this property

When we added the `hasMany` relation using the `lb4 relation` command, it added `the`
`@hasMany()` decorator  `which` defines this property . (Is this what we want to say?)

dhmlau replied:
actually, it added the property + the decorator. How about: 

When we added the `hasMany` relation using the `lb4 relation` command, it added the
`@hasMany()` decorator together with this property.

emonddr replied:
> actually, it added the property + the decorator. How about:
> 
> When we added the `hasMany` relation using the `lb4 relation` command, it added the
> `@hasMany()` decorator together with the `todo` property.

`todo` or `todos` ?  The line below says:

![image](https://user-images.githubusercontent.com/6864736/68954118-5a02b280-0791-11ea-9f43-671c1fb36dbf.png)

emonddr replied:
Also, we still need a `the` before '@hasMany decorator'

dhmlau replied:
it should be `todos` property (the one in the documentation is correct). 

OK. how about this:

When we added the `hasMany` relation using the `lb4 relation` command, it added
the `@hasMany()` decorator together with the `todos` property.

dhmlau commented:
@emonddr, please see my latest commit https://github.com/strongloop/loopback-next/pull/4126/commits/24e80aadf70764dddd8c959c7e37e3db43111a82 that addressed your comment. Thanks!

dhmlau commented:
@emonddr, I've pushed my changes and squashed all the commits. PTAL again. Thanks!
